spec.html

```diff
@@ -36455,28 +36456,28 @@ <h1>
         </emu-clause>
       </emu-clause>
 
-      <emu-clause id="sec-compilestring" type="sdo">
-        <h1>Runtime Semantics: CompileString</h1>
+      <emu-clause id="sec-compileclasssetstring" type="sdo">
+        <h1>Runtime Semantics: CompileClassSetString</h1>
         <dl class="header">
           <dt>description</dt>
-          <dd>It returns a string.</dd>
+          <dd>It returns a sequence of characters.</dd>
         </dl>
 
         <emu-grammar>ClassString :: [empty]</emu-grammar>
         <emu-alg>
-          1. Return the empty String.
+          1. Return an empty sequence of characters.
         </emu-alg>
         <emu-grammar>ClassString :: NonEmptyClassString</emu-grammar>
         <emu-alg>
-          1. Return CompileString of |NonEmptyClassString|.
+          1. Return CompileClassSetString of |NonEmptyClassString|.
         </emu-alg>
         <emu-grammar>NonEmptyClassString :: ClassCharacter NonEmptyClassString?</emu-grammar>
         <emu-alg>
           1. Let _cs_ be CompileToCharSet of |ClassCharacter|.
-          1. Let _s1_ be the string that is the single element of _cs_.
+          1. Let _s1_ be the sequence of characters that is the single element of _cs_.
           1. If |NonEmptyClassString| is present, then
-            1. Let _s2_ be CompileString of |NonEmptyClassString|.
-            1. Return the string that is the concatenation of _s1_ with _s2_.
+            1. Let _s2_ be CompileClassSetString of |NonEmptyClassString|.
+            1. Return the string-concatenation of _s1_ and _s2_.
```

Comment:
Oh, this doesn't work because the typing is wrong. The CharSet _cs_ is a mathematical set in which each element is a CharSetElement, which because _UnicodeSets_ is *true* is a sequence whose elements are [regular expression pattern] characters. So _s1_ is not a String value, a code unit, or a sequence of code units, and therefore cannot be used with [string-concatenation](https://tc39.es/ecma262/multipage/ecmascript-data-types-and-values.html#string-concatenation).

```suggestion
            1. Return the concatenation of _s1_ and _s2_.
```